Added Knockback

SurfsideRaven opened the pull request:
Enemies are now knocked back when hit with a damage that is less than their total health.

-Each unique enemy is able to set their specific modifier for the knockback distance, using the 'knockback_modifier' variable. This variable is set in each of the enemies files individually.

SurfsideRaven committed:
Added Knockback

Enemies are now knocked back when hit with a damage that is less than their total health.

AnthonyDrow21 commented on Scenes/NPC/Brute/EnemyBrute.gd:
```diff
@@ -10,12 +10,15 @@ func _physics_process(_delta):
 	move_and_slide()
 
-func DamageEnemy(damage):
+func DamageEnemy(damage, incoming_dmg_pos):
```
I think in these derived "DamageEnemy" functions we can use the "super" keyword to call the base enemy function. All the sub function need to do is define their knockback value and then call super.DamageEnemy(damage, position, knockback)

AnthonyDrow21 commented on Scenes/NPC/Enemy/Enemy.gd:
```diff
@@ -14,12 +14,15 @@ func _physics_process(_delta):
 	move_and_slide()
 
-func DamageEnemy(damage):
+func DamageEnemy(damage, incoming_dmg_pos):
```
I think we can add the knockback value here as an agrument and then have all the sub functions call this one.

SurfsideRaven committed:
Moved Knockback into Base Enemy Class

Created a default knockback values, which can be changed in the individual enemies.

SurfsideRaven committed:
Removing Temp Files

SurfsideRaven committed:
Merge branch 'main' into KnockbackEnemies
